templates/flavors/base/cluster-template.yaml

```diff
@@ -21,12 +21,12 @@ kind: AzureCluster
 metadata:
   name: ${CLUSTER_NAME}
 spec:
-  resourceGroup: "${AZURE_RESOURCE_GROUP:=$CLUSTER_NAME}"
+  resourceGroup: "${AZURE_RESOURCE_GROUP:=${CLUSTER_NAME}}"
   location: "${AZURE_LOCATION}"
   subscriptionID: ${AZURE_SUBSCRIPTION_ID}
   networkSpec:
     vnet:
-      name: "${AZURE_VNET_NAME:=$CLUSTER_NAME-vnet}"
+      name: "${AZURE_VNET_NAME:=${CLUSTER_NAME}-vnet}"
```

Comment:
why have this in the spec at all any more with webhook defaulting? should we consider slimming down the examples?

Reply:
Agreed, however I'd like to patch the release templates with minimal changes so I'd prefer merging this as is and then thinking about reducing the number of variables in the template. Also I think we shouldn't do this before #802 since the vnetName still needs to be consistent with azure.json and exposing one but not the other could lead to issues.